### PR TITLE
Qwen3-VL fused loop for language and vision model

### PR DESCRIPTION
**Pull request by ssanjayTT**
### Summary
Currently, prefill first loops over all users for image processing, stores intermediate vision tokens and deepstack_embeds in DRAM, then preprocesses (merges and scatters) the vision tokens with the text embeds and runs the text backbone which also loops over all users. This approach has a few problems:
1. It causes hangs in larger batches in the vllm tt-inference-server runs (probably due to these large tensors being kept in memory)
2. It has poor DRAM utilization since all users 
3. It prevents the possibility of introducing end to end trace capture on the entire model.
4. Introduces batch concatenation and other processing which might hurt performance.

We replace this with a fused loop in generator vllm which essentially processes a single user at a time. It runs the vision model, merges vision tokens and deep-stack embeds with text embeds, runs the text backbone on that user and extracts the logits. This improves the DRAM utilization of the model, prevents hangs and introduces the possibility of introducing trace capture on the model.

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:ssanjay/qwen3-vl-fused-gen-loop)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:ssanjay/qwen3-vl-fused-gen-loop)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:ssanjay/qwen3-vl-fused-gen-loop)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:ssanjay/qwen3-vl-fused-gen-loop)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:ssanjay/qwen3-vl-fused-gen-loop)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:ssanjay/qwen3-vl-fused-gen-loop)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:ssanjay/qwen3-vl-fused-gen-loop)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:ssanjay/qwen3-vl-fused-gen-loop)

### CODE REVIEW

**Review by gwangTT — APPROVED** [review]
Looking forward to performance numbers!

**Comment by ssanjayTT** [comment]
https://github.com/tenstorrent/tt-metal/actions/runs/25591560465/job/75130715857 demo tests passing
